### DISCUSSION TITLE
test with dynamic data?

### DISCUSSION THREAD

**Discussion by lucy7099** [percy/cli#504] [Support] [answered]
is there any way to test pages with dynamic data that can cause diffs when the data changes between snapshots?

i think with percyscript you can do this by intercepting the network request and sending a mocked response instead. But is it possible to do this with the snapshot CLI

thanks

**Comment by Robdel12** [comment, marked as answer]
Hey @lucy7099! 

> i think with percyscript you can do this by intercepting the network request and sending a mocked response instead

For something like that you'll want to use the Puppeteer SDK: https://docs.percy.io/docs/puppeteer The PercyScript SDK was a _really_ thin wrapper around Puppeteer, so the same method would translate 😃

**Reply by cancan101** [reply]
Similar question for a more constrained use case, is there any way to freeze the datetime used in testing?

**Reply by wwilsman** [reply]
With the snapshot command, you'd have to inject javascript into the page to freeze/adjust the dates. For SDKs that integrate with test frameworks, there are some mocking libraries available: https://docs.percy.io/docs/freezing-dynamic-data

**Reply by cancan101** [reply]
I ended up using the [`faketime` pre-loader](https://github.com/wolfcw/libfaketime) to solve this problem. I am able to install on github actions with a simple `apt-get install faketime`.

Then I can just run:
```
faketime 2021-08-17 20:30:00 yarn ci:visual-regression
```
